tests/test_unittest/test_gtc/test_gtcpp_compilation.py

```diff
@@ -0,0 +1,160 @@
+import re
+from pathlib import Path
+from typing import Literal, Pattern, Union
+
+import pytest
+import setuptools
+from devtools import debug
+
+from gt4py import config, gt2_src_manager  # TODO must not include gt4py package or ok for test?
+from gt4py.gtc.gtcpp.gtcpp import AssignStmt, GTAccessor, GTApplyMethod, GTExtent, Intent, Program
+from gt4py.gtc.gtcpp.gtcpp_codegen import GTCppCodegen
+from gt4py.gtc.gtcpp.oir_to_gtcpp import _extract_accessors
+
+from .gtcpp_utils import AccessorRefBuilder, GTApplyMethodBuilder, GTFunctorBuilder, ProgramBuilder
+
+
+if not gt2_src_manager.has_gt_sources() and not gt2_src_manager.install_gt_sources():
+    raise RuntimeError("Missing GridTools sources.")
+
+
+# TODO does this exist in pytest already?
+def match(value: str, regexp: "Union[str, Pattern]") -> "Literal[True]":
+    """
+    Stolen from `pytest.raises`.
+    Check whether the regular expression `regexp` matches `value` using :func:`python:re.search`.
+    If it matches `True` is returned.
+    If it doesn't match an `AssertionError` is raised.
+    """
+    assert re.search(regexp, str(value)), "Pattern {!r} does not match {!r}".format(
+        regexp, str(value)
+    )
+    return True
+
+
+# TODO can I get tmp_path automagically here?
```

Comment:
yes, by making it a fixture that returns the function

Reply:
left it for now as the indirection doesn't simplify the code